x/programs/rust/examples/token/src/lib.rs

```diff
@@ -63,6 +63,16 @@ pub fn mint_to(program: Program, recipient: Address, amount: i64) -> bool {
     true
 }
 
+/// Burn the token from the recipient.
+#[public]
+pub fn burn_from(program: Program, recipient: Address) -> bool {
+    program
+        .state()
+        .delete(StateKey::Balance(recipient))
+        .expect("failed to burn recipient tokens");
+    true
+}
```

Comment:
Can you add a test here? This is kind of an odd example IMO, usually you would burn a specific amount of tokens, not the entire balance. 

A simulator test makes the intended behaviour crystal clear

Reply:
I can rename it as `burn_all` or `burn_total` if you prefer.
It is mostly to test on golang side the usage of the `delete` function.

Reply:
Don't worry about the rename, thanks for adding the test!